### DISCUSSION TITLE
What are the differences between using an API Route and using a mutation?

### DISCUSSION THREAD

**Discussion by zwl1619** [blitz-js/blitz#2270] [Q&A] [answered]
I want to upload a file to my server, there are two ways:
1、using an API Route 
2、use a mutation
What are the differences between the two methods above?

**Comment by MrLeebo** [comment, marked as answer]
For a file upload you would use an API route (or getServerSideProps). Mutations can only send and receive JSON as the content type, but you probably want to upload the file using multipart form data instead.
